### PR TITLE
chore(docs): compound 8 learnings from yellow-council Wave-2 review (PRs #328-#330)

### PR DESCRIPTION
**Pull request by KingInYellow18**
5 new solution docs + 3 amendments + MEMORY.md pointers covering: awk PEM state machine variable mutation, $(cat $FILE) ARG_MAX antipattern, prompt-injection fence delimiter escape, shell &&/|| operator precedence bugs, jq select filter for JSONL, SKILL user-invokable presence rule, README plugin count drift, bash-block status/summary variable scope.

## Summary by CodeRabbit

* **Documentation**
  * Added guides addressing code-quality anti-patterns: bash subprocess isolation, command substitution size limits, jq JSONL filtering, and shell operator precedence.
  * Added security documentation covering PEM redaction state machine failures and prompt-injection fence delimiter vulnerabilities.
  * Enhanced documentation for frontmatter and README validation requirements.

<h3>Greptile Summary</h3>

This PR compounds eight engineering learnings from yellow-council Wave-2 review (PRs #328–#330) into the `docs/solutions` knowledge base: five new solution docs and three amendments to existing ones, covering correctness, reliability, and security patterns in shell/awk/jq command-file authoring.

- **5 new docs** added: `$(cat)` ARG_MAX antipattern, `jq select()` vs `first(expr // empty)` for JSONL, `&&`/`||` operator-precedence guard bugs, awk PEM state-machine variable-mutation (security), and prompt-injection fence-delimiter escape (security).
- **3 amendments** added to existing docs: bash subshell isolation (status/summary block concretization), skill frontmatter `user-invokable` presence rule, and README plugin-count / directory-tree drift (including a CI enforcement script skeleton).

<h3>Confidence Score: 5/5</h3>

All changes are documentation only; no production code is modified.

Every changed file is a Markdown solution doc or amendment. The technical analysis across all new docs is accurate. The two findings are both minor issues confined to the CI example script within one doc and affect no running code.

docs/solutions/code-quality/stale-env-var-docs-and-prose-count-drift.md — the bash quick-check and the CI script skeleton use different counting methods for plugin components.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| docs/solutions/code-quality/bash-block-subshell-isolation-in-command-files.md | Amendment: adds Status/Summary Blocks concretization of cross-block variable loss with sentinel-file, state-file, and merge-block fixes; technically accurate. |
| docs/solutions/code-quality/cat-file-argmax-inline-substitution-antipattern.md | New doc: documents the $(cat FILE) ARG_MAX antipattern with two independent bugs (dead stdin pipe, per-argument kernel limit); fixes and detection are correct. |
| docs/solutions/code-quality/jq-select-filter-over-first-empty-for-jsonl.md | New doc: accurately describes the first(expr // empty) empty-generator error on JSONL streams and the select() replacement pattern; examples are correct. |
| docs/solutions/code-quality/shell-and-or-chain-operator-precedence-bugs.md | New doc: correctly documents the left-associative && / || precedence hazard in guard-then-act-then-warn chains and provides the if/then/fi fix. |
| docs/solutions/code-quality/skill-frontmatter-attribute-and-format-requirements.md | Amendment: adds rule requiring user-invokable field presence for internal skills, with detection script and on-touch extension; no issues. |
| docs/solutions/code-quality/stale-env-var-docs-and-prose-count-drift.md | Amendment: adds plugin-count drift rule with bash quick-check and JS CI skeleton; bash uses ls | wc -l (counts subdirs/non-md files) while CI script filters to .md only — inconsistent counts across the same doc. |
| docs/solutions/security-issues/awk-pem-state-machine-variable-mutation.md | New doc: correctly identifies the mutate-before-test bug in the PEM awk state machine and the fix (test $0, derive line after state update); single-line PEM edge case handled correctly. |
| docs/solutions/security-issues/prompt-injection-fence-delimiter-escape.md | New doc: accurately documents the static-fence-close-tag escape vulnerability and three mitigations (nonce, pre-fencing sweep, base64-armor); caveats for Option A are correctly noted. |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[yellow-council Wave-2 PRs 328-330] --> B[5 new solution docs]
    A --> C[3 doc amendments]
    B --> D[cat-file-argmax antipattern]
    B --> E[jq-select-filter for JSONL]
    B --> F[shell and-or chain precedence bugs]
    B --> G[awk-pem state machine]
    B --> H[prompt-injection fence escape]
    C --> I[bash-block subshell isolation]
    C --> J[skill-frontmatter requirements]
    C --> K[stale docs and count drift]
    G --> L[Fix: test $0 not mutated variable]
    H --> M[Fix: nonce or sweep or base64]
    E --> N[Fix: select plus head -1]
    F --> O[Fix: if/then/fi explicit branch]
    K --> P[Fix: scripts/check-readme-count.js]
```

<a href="https://chatgpt.com/codex/deeplink?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22kinginyellows%2Fyellow-plugins%22%20on%20the%20existing%20branch%20%22chore%2Fcompound-yellow-council-learnings%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22chore%2Fcompound-yellow-council-learnings%22.%0A%0AFix%20the%20following%202%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%202%0Adocs%2Fsolutions%2Fcode-quality%2Fstale-env-var-docs-and-prose-count-drift.md%3A258-261%0AThe%20bash%20quick-check%20uses%20plain%20%60ls%20%7C%20wc%20-l%60%2C%20which%20counts%20every%20entry%20in%20the%20directory%20%E2%80%94%20including%20subdirectories%20and%20non-Markdown%20files%20%E2%80%94%20while%20the%20CI%20skeleton%20immediately%20below%20it%20filters%20to%20%60.endsWith%28'.md'%29%60.%20A%20%60commands%2F%60%20directory%20containing%20a%20subdirectory%20will%20produce%20a%20higher%20count%20here%20than%20CI%20does%2C%20generating%20a%20false%20mismatch%20that%20the%20doc%20itself%20calls%20a%20required%20fix%20before%20merging.%20Aligning%20the%20bash%20check%20to%20the%20same%20%60.md%60-only%20criterion%20avoids%20confusion.%0A%0A%60%60%60suggestion%0A%23%20Check%20filesystem%20counts%20%28.md%20files%20only%20%E2%80%94%20matches%20CI%20script%20and%20README%20table%29%0Aecho%20%22commands%3A%20%24%28ls%20%22%24GIT_ROOT%2Fplugins%2F%24PLUGIN%2Fcommands%2F%22*.md%202%3E%2Fdev%2Fnull%20%7C%20wc%20-l%29%22%0Aecho%20%22agents%3A%20%20%20%24%28ls%20%22%24GIT_ROOT%2Fplugins%2F%24PLUGIN%2Fagents%2F%22*.md%20%20%202%3E%2Fdev%2Fnull%20%7C%20wc%20-l%29%22%0Aecho%20%22skills%3A%20%20%20%24%28ls%20%22%24GIT_ROOT%2Fplugins%2F%24PLUGIN%2Fskills%2F%22*.md%20%20%202%3E%2Fdev%2Fnull%20%7C%20wc%20-l%29%22%0A%60%60%60%0A%0A%23%23%23%20Issue%202%20of%202%0Adocs%2Fsolutions%2Fcode-quality%2Fstale-env-var-docs-and-prose-count-drift.md%3A303%0APlugin%20names%20are%20interpolated%20directly%20into%20a%20%60RegExp%60%20constructor%20without%20escaping.%20Kebab-case%20names%20like%20%60yellow-browser-test%60%20are%20safe%20today%2C%20but%20a%20name%20containing%20%60.%60%20would%20make%20it%20match%20any%20character%20and%20potentially%20report%20a%20false%20count%20against%20the%20wrong%20row.%20Since%20this%20is%20a%20CI%20skeleton%20intended%20for%20future%20adaptation%2C%20escaping%20%60plugin%60%20before%20constructing%20the%20regex%20guards%20against%20silent%20mismatches.%0A%0A%60%60%60suggestion%0A%20%20%20%20const%20escapedPlugin%20%3D%20plugin.replace%28%2F%5B.*%2B%3F%5E%24%7B%7D%28%29%7C%5B%5C%5D%5C%5C%5D%2Fg%2C%20'%5C%5C%24%26'%29%3B%0A%20%20%20%20const%20rowMatch%20%3D%20readme.match%28new%20RegExp%28%60%24%7BescapedPlugin%7D%5B%5E%5C%5Cn%5D*%5C%5Cb%28%5C%5Cd%2B%29%5C%5Cs%2B%24%7Bsubdir%7D%60%29%29%3B%0A%60%60%60%0A%0A"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodexDark.svg?v=2"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodex.svg?v=2">Fix All in Codex</picture></a>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
docs/solutions/code-quality/stale-env-var-docs-and-prose-count-drift.md:258-261
The bash quick-check uses plain `ls | wc -l`, which counts every entry in the directory — including subdirectories and non-Markdown files — while the CI skeleton immediately below it filters to `.endsWith('.md')`. A `commands/` directory containing a subdirectory will produce a higher count here than CI does, generating a false mismatch that the doc itself calls a required fix before merging. Aligning the bash check to the same `.md`-only criterion avoids confusion.

```suggestion
# Check filesystem counts (.md files only — matches CI script and README table)
echo "commands: $(ls "$GIT_ROOT/plugins/$PLUGIN/commands/"*.md 2>/dev/null | wc -l)"
echo "agents:   $(ls "$GIT_ROOT/plugins/$PLUGIN/agents/"*.md   2>/dev/null | wc -l)"
echo "skills:   $(ls "$GIT_ROOT/plugins/$PLUGIN/skills/"*.md   2>/dev/null | wc -l)"
```

### Issue 2 of 2
docs/solutions/code-quality/stale-env-var-docs-and-prose-count-drift.md:303
Plugin names are interpolated directly into a `RegExp` constructor without escaping. Kebab-case names like `yellow-browser-test` are safe today, but a name containing `.` would make it match any character and potentially report a false count against the wrong row. Since this is a CI skeleton intended for future adaptation, escaping `plugin` before constructing the regex guards against silent mismatches.

```suggestion
    const escapedPlugin = plugin.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
    const rowMatch = readme.match(new RegExp(`${escapedPlugin}[^\\n]*\\b(\\d+)\\s+${subdir}`));
```

`````

</details>

<sub>Reviews (3): Last reviewed commit: ["fix(docs): clear PR-introduced markdownl..."](https://github.com/kinginyellows/yellow-plugins/commit/d23eb315cdb08620283451ef66061f7d70330710) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30734451)</sub>

### CODE REVIEW

**Comment by KingInYellow18** [comment]
* **#331** <a href="https://app.graphite.com/github/pr/KingInYellows/yellow-plugins/331?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/KingInYellows/yellow-plugins/331?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/KingInYellows?ref=merge-queue-instructions-in-stack-comment&prId=331).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: docs/solutions/code-quality/skill-frontmatter-attribute-and-format-requirements.md [hunk 1]
```diff
@@ -293,3 +293,50 @@ special characters predictably.
...
+not the requirement that the field must be present for internal skills.
+
+**The gap:** The peer pattern — `codex-patterns`, `pr-review-workflow`,
+`debt-conventions` — all carry `user-invokable: false` for internal skills.
+Without the field, Claude Code's frontmatter parser applies an undefined
+default, and CI tooling (including automated reviewers) flags the omission.
+
+**Rule:** Every internal SKILL.md (one not intended for direct user invocation)
+MUST include `user-invokable: false` in its frontmatter, regardless of whether
+the skill has any other frontmatter.
```
[the remark this answers is not in the document]
Comment (KingInYellow18) [reply docs/solutions/code-quality/skill-frontmatter-attribute-and-format-requirements.md:315]: False positive — verified locally.

Line 313: 79 chars (`**Rule:** Every internal SKILL.md (one not intended for direct user invocation)`)
Line 314: 78 chars (`MUST include \`user-invokable: false\` in its frontmatter, regardless of whether`)

Both are within the 80-char limit. The project's `.markdownlint.json` actually sets `MD013.line_length: 120`, so this wouldn't trip even at 81 chars. No change needed.

---
_Generated by [Claude Code](https://claude.ai/code)_
